# MERCI: tracing of mitochondrion transfer between single cancer and T cells
MERCI contains two modules:MERCI-mtSNP.py for calling mtSNVs and MERCI R package for predicting the mitochondrial recipient cells and their mitochondrial compositions. 

#usage of MERCI-mtSNP

**MERCI-mtSNP is written in Python3, with the following dependencies: 

- pandas
- numpy
- pysam
- matplotlib

Please make sure all dependency modules are installed before usiung MERCI-mtSNP.

## Usage
**Type python MERCI-mtSNP.py -h to display all the commandline options:**

| Parameters | Description |
| ------------- | ------------- |
| -h, --help | show this help message and exit |
| -D DATATYPE, --dataType=DATATYPE | The data type of sequencing data. One of '10x_scRNA-seq'(default), 'smart-seq2', 'bulk_ATAC-seq', 'scATAC-seq', or 'bulk_RNA-seq'. |
| -o DIRECTORY, --output=DIRECTORY | Output directory for intermediate and final outputs. |
| -S SAMPLEID , --sampleID=SAMPLEID | the sample identifier, also serve as the prefix of output file. if not given, the names of all intermediate or final output files will be automatically set as sampleX. |
| -b PATH_BAM, --Bamfile=PATH_BAM | Input bam file for MT mutation calling |
| -f PATH_FA, --fastafile=PATH_FA | The fasta data of genome reference sequence in the used reference genome file, usually named as XXX.fa |
| -c PATH_BARCODES, --CellBarcode=PATH_BARCODES | Only work for 10x_scRNA-seq data type, the directory where cell barcodes file (barcodes.tsv.gz or barcodes.tsv) generated by cell ranger exists in |
| -M MQUALITY, --MQcutoff=MQUALITY | The lowest alignment quality that are accepted, the reads with alignment scores below the given value will be discarded, default=5 for scATAC-seq or bulk ATAC-seq, default=255 for other dataTypes |
| -B QCUTOFF, --BQcutoff=QCUTOFF | The base quality cutoff, only alleles with BQ higher than this value will be retained, default=15 for 10x_scRNA-seq, default=25 for other dataTypes |
| -l LN, --ln=LN | Only work for 10x_scRNA-seq data type, the maximum length of genomic region for SNP clusters, reads supporting multiple variants within a small genomic region (ln bp) will be removed, default=5 |
| -m MINC, --minC=MINC | For all data types except 10x_scRNA-seq, A threshold for coverage, the faction of MT genome that was covered by reads no less than this value will be recorded on the generated coverage figure, default=1 |





















