#' read in mtSNV of single cell sequencing data
#'
#' This function read in MT variants table (*.MT_variants.txt) generated by MERCI-mtSNP, works for data types: smart-seq2, bulk RNA-seq, scATAC-seq, bulk ATAC-seq.
#' @param varFile The path to .MT_variants.txt file generated by MERCI-mtSNP.
#' @param cellname  A character reprents the cell indentifier.
#' @return the mtSNV annotation table
#' @export
#' @examples readMTvar(varFile='./X.MT_variants.txt', cellname='XXX')

readMTvar <- function(varFile, cellname='XXX')
{
	MT_variants <- read.csv(varFile, header=TRUE, sep="\t", fill=TRUE, quote="", check.names=FALSE, row.names = NULL) ;
	MT_variants <- data.frame(Cell=cellname, MT_variants[, c('ID', 'pos', 's_reads', 't_reads', 'AF', 'avg_BQ')], stringsAsFactors=FALSE) ;
	
	return(MT_variants) ;
}

